core-api-library/library/src/main/java/net/gini/android/core/api/authorization/UserRepository.kt

```diff
@@ -55,6 +52,15 @@ class UserRepository(
                 emit(loginClientForSession().firstOrNull())
             }
         }
+
+    suspend fun updateEmail(newEmail: String, oldEmail: String, session: SessionToken) {
+        getUserRepositorySession().collect {
```

Comment:
I am wondering why `getUserRepositorySession` returns a flow 🤔. Currently it's return a cold flow emitting only one value so wouldn't it be simpler to just have it return `Resource<SessionToken>.

Same applies to the other method returning flows. I can't think of any advantage of using flows here. Am I missing something? 🤔